packages/orchestration/src/withdraw-reward.js

```diff
@@ -0,0 +1,72 @@
+// @ts-check
+
+// question: are we doing snake-case filenames?
+
+import { E } from '@endo/far';
+import {
+  MsgWithdrawDelegatorReward,
+  MsgWithdrawDelegatorRewardResponse,
+} from '@agoric/cosmic-proto/cosmos/distribution/v1beta1/tx.js';
+import { Any } from '@agoric/cosmic-proto/google/protobuf/any';
+import { decodeBase64 } from '@endo/base64';
+
+const { Fail } = assert;
+
+/**
+ * @import { ChainAccount } from './types.js';
+ * @import {AnyJson} from '@agoric/cosmic-proto';
+ */
+
+/** @param {{ delegatorAddress: string, validatorAddress: string}} fields */
+export const makeWithdrawDelegatorRewardProtoMsg = ({
+  delegatorAddress,
+  validatorAddress,
+}) =>
+  MsgWithdrawDelegatorReward.toProtoMsg({
+    delegatorAddress,
+    validatorAddress,
+  });
+
+// XXX why doesn't Any.toJSON return AnyJson???
+/**
+ * @param {Any} x
+ * @returns {AnyJson}
+ */
+const toJSON = x => /** @type {AnyJson} */ (Any.toJSON(x));
+
+// ???: issue / spec calls this WithdrawDelegatorReward
+// am I right to take that literally? don't most of our function
+// names start with lower-case?
+
+/**
+ * Execute transaction to withdraw delegator rewards from a validator.
+ *
+ * @param {ERef<ChainAccount>} account
+ * @param {string} validatorAddress
+ */
+export const WithdrawDelegatorReward = async (account, validatorAddress) => {
+  // XXX should account be remote(able)? or must it be local?
+  const delegatorAddress = await E(account).getAccountAddress();
+
+  const msg0 = makeWithdrawDelegatorRewardProtoMsg({
+    delegatorAddress,
+    validatorAddress,
+  });
+  const ackStr = await E(account).executeEncodedTx([toJSON(msg0)]);
+  //   ackStr || Fail`unexpected empty reply to WithdrawDelegatorReward`;
+
+  // XXX should decodeBase64 come from endo or from cosmjs?
```

Comment:
`endo`. Also - `Msg*.toJSON()` and `Msg*.fromJSON()` exports from `@agoric/cosmic-proto` are patched to use `decodeBase64` and `encodeBase64` from `@endo/base64`.

Reply:
why?

Reply:
cosmic-proto uses Telescope to codegen and that was looking for `atob` in the globals, which was failing.